W3ChampionsStatisticService/Hubs/ConnectionMapping.cs

```diff
@@ -6,40 +6,67 @@ namespace W3ChampionsStatisticService.Hubs;
 public class ConnectionMapping
 {
     private readonly Dictionary<string, WebSocketUser> _connections = new();
+    private readonly Dictionary<string, HashSet<string>> _onlineUsersByBattleTag = new();
+    private readonly object _lock = new();
 
     public List<WebSocketUser> GetUsers()
     {
-        lock (_connections)
+        lock (_lock)
         {
             return _connections.Values.Select(v => v).OrderBy(r => r.BattleTag).ToList();
         }
     }
 
     public void Add(string connectionId, WebSocketUser user)
     {
-        lock (_connections)
+        lock (_lock)
         {
             if (!_connections.ContainsKey(connectionId))
             {
                 _connections.Add(connectionId, user);
+                if (!_onlineUsersByBattleTag.ContainsKey(user.BattleTag))
+                {
+                    _onlineUsersByBattleTag[user.BattleTag] = new HashSet<string>();
+                }
+                _onlineUsersByBattleTag[user.BattleTag].Add(connectionId);
             }
         }
     }
 
     public WebSocketUser GetUser(string connectionId)
     {
-        lock (_connections)
+        lock (_lock)
```

Comment:
Hmm, locking on every get seems tough. Could you check if we have an easy way to get a read write lock?

Reply:
Updated